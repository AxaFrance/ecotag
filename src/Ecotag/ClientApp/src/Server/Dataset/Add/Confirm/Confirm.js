import React from 'react';
import Title from '../../../../TitleBar';
import Stepper from '../../../shared/Stepper';
import Button from '@axa-fr/react-toolkit-button';
import {withRouter} from 'react-router-dom';
import './Confirm.scss';
import useProjectTranslation from "../../../../translations/useProjectTranslation";

export const Confirm = ({navBack}) => {
    const {translate} = useProjectTranslation();

    return(
        <>
            <Title title={translate('dataset.confirm.title')} goTo="/datasets" goTitle="Datasets"/>
            <Stepper activeStep="confirm" title={translate('dataset.confirm.stepper')} link="/datasets/new"/>
            <div className="af-confirm">
                <span className="glyphicon glyphicon--ring glyphicon-ok"/>
                <h2 className="af-confirm__message">{translate('dataset.confirm.confirmation_title')}</h2>
                <h4 className="af-confirm__message">{translate('dataset.confirm.confirmation_description')}</h4>
                <Button classModifier="success hasiconLeft" id="return_datatset" name="return_datatset" onClick={navBack}>
                    <span className="af-btn-text">{translate('dataset.confirm.return_button_label')}</span>
                    <span className="icons-list__item-icon glyphicon glyphicon-list-alt"/>
                </Button>
            </div>
        </>
    );
}

const ConfirmContainer = ({history}) => {
    const navBack = () => {
        history.push('/datasets');
    };
    return <Confirm navBack={navBack}/>;
};

export default withRouter(ConfirmContainer);
