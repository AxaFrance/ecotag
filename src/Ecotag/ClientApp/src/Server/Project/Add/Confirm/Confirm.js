import React from 'react';
import Title from '../../../../TitleBar';
import Stepper from '../../../shared/Stepper';
import Button from '@axa-fr/react-toolkit-button';
import {useLocation, withRouter} from 'react-router-dom';
import './Confirm.scss';
import useProjectTranslation from '../../../../translations/useProjectTranslation';

export const Confirm = ({navBack, navViewProject}) => {
    const {translate} = useProjectTranslation();

    return(
        <>
            <Title title={translate('project.confirm.title')} goTo="/projects" goTitle="Projets"/>
            <Stepper activeStep="confirm" title={translate('project.confirm.stepper')} link="/projects/new"/>
            <div className="af-confirm">
                <span className="glyphicon glyphicon--ring glyphicon-ok"/>
                <h2 className="af-confirm__message">{translate('project.confirm.new_project_added')}</h2>
                <Button classModifier="success hasiconLeft" id="view_project" name="view_project" onClick={navViewProject}>
                    <span className="af-btn-text">{translate('project.confirm.view_project_button_label')}</span>
                    <span className="icons-list__item-icon glyphicon glyphicon-list-alt"/>
                </Button>
                <Button classModifier="success hasiconLeft" id="return_datatset" name="return_datatset" onClick={navBack}>
                    <span className="af-btn-text">{translate('project.confirm.go_back_button_label')}</span>
                    <span className="icons-list__item-icon glyphicon glyphicon-list-alt"/>
                </Button>
            </div>
        </>
    );
}

export const ConfirmContainer = ({history}) => {
    const location = useLocation();
    const projectId = location.state.projectId;
    const navBack = () => {
        history.push('/projects');
    };
    const navViewProject = () => {
        history.push(`${projectId}`);
    };
    return <Confirm navBack={navBack} navViewProject={navViewProject}/>;
};

export default withRouter(ConfirmContainer);
