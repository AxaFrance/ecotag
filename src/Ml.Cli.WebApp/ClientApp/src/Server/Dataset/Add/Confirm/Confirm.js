import React from 'react';
import Title from '../../../../TitleBar';
import Stepper from '../../../shared/Stepper';
import Button from '@axa-fr/react-toolkit-button';
import {useLocation, withRouter} from 'react-router-dom';
import './Confirm.scss';

export const Confirm = ({ navBack }) => {
    
    const location = useLocation();
    const filesResult = location.state.filesResult;
    const koFilesArray = Object.keys(filesResult).filter(key => filesResult[key] !== null);
    const containsKOFiles = koFilesArray.length > 0;
    
    return(
        <>
            <Title title="Confirmation" goTo="/datasets" goTitle="Datasets" />
            <Stepper activeStep="confirm" title="Nouveau dataset" link="/datasets/new" />
            <div className="af-confirm">
                <span className="glyphicon glyphicon--ring glyphicon-ok" />
                <h2 className="af-confirm__message">Nouveau dataset ajouté !</h2>
                {containsKOFiles &&
                    <div className="af-danger">
                        <h4>
                            Certains fichiers n'ont pas été ajoutés au dataset :
                        </h4>
                        {koFilesArray.map((koFileName, index) => (
                            <div key={index}>
                                <span className="af-danger__message">{koFileName} - {filesResult[koFileName]}</span>
                            </div>
                        ))}
                    </div>
                }
                <Button classModifier="success hasiconLeft" id="return_datatset" name="return_datatset" onClick={navBack}>
                    <span className="af-btn-text">Retour liste datasets</span>
                    <span className="icons-list__item-icon glyphicon glyphicon-list-alt" />
                </Button>
            </div>
        </>
    )
};

const ConfirmContainer = ({ history }) => {
  const navBack = () => {
    history.push('/datasets');
  };
  return <Confirm navBack={navBack} />;
};

export default withRouter(ConfirmContainer);
