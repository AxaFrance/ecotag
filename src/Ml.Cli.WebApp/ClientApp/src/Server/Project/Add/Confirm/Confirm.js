import React from 'react';
import Title from 'TitleBar';
import Stepper from '../../../shared/Stepper';
import Button from '@axa-fr/react-toolkit-button';
import { useLocation, withRouter } from 'react-router-dom';
import './Confirm.scss';

export const Confirm = ({ navBack, navViewProject }) => (
  <>
    <Title title="Confirmation" goTo="/projects" goTitle="Projets" />
    <Stepper activeStep="confirm" title="Nouveau projet" link="/projects/new" />
    <div className="af-confirm">
      <span className="glyphicon glyphicon--ring glyphicon-ok" />
      <h2 className="af-confirm__message">Nouveau projet ajout√© !</h2>
      <Button classModifier="success hasiconLeft" id="view_project" name="view_project" onClick={navViewProject}>
        <span className="af-btn-text">Visualiser le projet</span>
        <span className="icons-list__item-icon glyphicon glyphicon-list-alt" />
      </Button>
      <Button classModifier="success hasiconLeft" id="return_datatset" name="return_datatset" onClick={navBack}>
        <span className="af-btn-text">Retour liste des projets</span>
        <span className="icons-list__item-icon glyphicon glyphicon-list-alt" />
      </Button>
    </div>
  </>
);

export const ConfirmContainer = ({ history }) => {
  const location = useLocation();
  const project = location.state.project;
  const navBack = () => {
    history.push('/');
  };
  const navViewProject = () => {
    history.push(`${project.id}`);
  };
  return <Confirm navBack={navBack} navViewProject={navViewProject} />;
};

export default withRouter(ConfirmContainer);
