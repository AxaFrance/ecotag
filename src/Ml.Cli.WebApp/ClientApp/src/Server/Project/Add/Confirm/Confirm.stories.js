import React from 'react';
import { action } from '@storybook/addon-actions';
import { Confirm } from './Confirm';
import { BrowserRouter as Router } from 'react-router-dom';

export default { title: 'Project/Add/Confirm' };

export const withDefault = () => <Router><Confirm onClick={action('onClick')} /></Router>;
