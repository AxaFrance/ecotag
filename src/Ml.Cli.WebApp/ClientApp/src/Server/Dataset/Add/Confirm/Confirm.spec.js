import {render} from "@testing-library/react";
import {BrowserRouter as Router} from "react-router-dom";
import React from "react";
import {Confirm} from "./Confirm";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        state: {filesResult: {firstFile: "InvalidFileExtension", secondFile: null, thirdFile: "InvalidFileExtension"}}
    })
}));

describe('Confirm', () => {
    it('Should render correctly', () => {
        const {container, asFragment} = render(<Router><Confirm fetch={fetch} /></Router>);
        expect(asFragment()).toMatchSnapshot();
    })
})