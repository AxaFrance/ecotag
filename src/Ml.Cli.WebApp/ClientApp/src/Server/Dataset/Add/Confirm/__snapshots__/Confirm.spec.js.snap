// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Should render correctly 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/datasets"
        title="Datasets"
      >
        <i
          class="glyphicon glyphicon-arrowthin-left"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Confirmation
      </h1>
    </div>
  </div>
  <div
    class="af-steps-new af-steps-new--devis"
  >
    <ol
      class="af-steps-list"
    >
      <li
        class="past af-steps-list-step number"
        title="Nouveau dataset"
      >
        <a
          class="af-steps-list-stepLabel"
          href="/datasets/new"
        >
          <div
            class="af-steps-list-stepNumber"
          >
            1
          </div>
          <div
            class="af-steps-list-stepTitle"
          >
            Nouveau dataset
          </div>
        </a>
      </li>
      <li
        class="af-steps-list-step af-steps-list-step--success"
        title="Confirmation"
      >
        <div
          class="af-steps-list-stepLabel"
        >
          <span
            class="af-steps-list-stepTitle"
          >
            Confirmation
          </span>
        </div>
      </li>
    </ol>
  </div>
  <div
    class="af-confirm"
  >
    <span
      class="glyphicon glyphicon--ring glyphicon-ok"
    />
    <h2
      class="af-confirm__message"
    >
      Nouveau dataset ajouté !
    </h2>
    <div
      class="af-danger"
    >
      <h4>
        Certains fichiers n'ont pas été ajoutés au dataset :
      </h4>
      <div>
        <span
          class="af-danger__message"
        >
          firstFile - InvalidFileExtension
        </span>
      </div>
      <div>
        <span
          class="af-danger__message"
        >
          thirdFile - InvalidFileExtension
        </span>
      </div>
    </div>
    <button
      class="btn af-btn af-btn--success af-btn--hasiconLeft"
      id="return_datatset"
      name="return_datatset"
    >
      <span
        class="af-btn-text"
      >
        Retour liste datasets
      </span>
      <span
        class="icons-list__item-icon glyphicon glyphicon-list-alt"
      />
    </button>
  </div>
</DocumentFragment>
`;
